### PR TITLE
Add nicer restart script

### LINKED ISSUE #1320: Automated Restart Crashes Server
**Describe the bug**
The Server Occasionally automatically restarts and does not come back. A restart of the systemd service fixes the problem. 

**To Reproduce**
1. Install Jellyfin from Ubuntu (or Debian) repository
2. Let run for a period
3. On idle Service will have attempted to restart and be stopped

**Expected behavior**
Service should continue running after restart.

**Logs**
```
[2019-04-27 11:30:41.728 +00:00] [INF] App needs to be restarted.
[2019-04-27 11:30:41.731 +00:00] [INF] "Check for plugin updates" Completed after 0 minute(s) and 1 seconds
[2019-04-27 11:30:41.774 +00:00] [INF] ExecuteQueuedTasks
[2019-04-27 11:45:41.749 +00:00] [INF] Automatically restarting the system because it is idle and a restart is required.
[2019-04-27 11:45:41.751 +00:00] [INF] Calling RestartInternal
[2019-04-27 11:45:41.753 +00:00] [INF] Disposing "CoreAppHost"
[2019-04-27 11:45:41.754 +00:00] [INF] Disposing "ActivityLogWebSocketListener"
[2019-04-27 11:45:41.755 +00:00] [INF] Disposing "SessionInfoWebSocketListener"
[2019-04-27 11:45:41.757 +00:00] [INF] Disposing "ScheduledTasksWebSocketListener"
[2019-04-27 11:45:41.757 +00:00] [INF] Disposing "SessionWebSocketListener"
[2019-04-27 11:45:41.758 +00:00] [INF] Disposing "EmbyTV"
[2019-04-27 11:45:41.758 +00:00] [INF] Disposing "ApiEntryPoint"
[2019-04-27 11:45:41.759 +00:00] [INF] Disposing "ServerEntryPoint"
[2019-04-27 11:45:41.759 +00:00] [INF] Disposing "EntryPoint"
[2019-04-27 11:45:41.759 +00:00] [INF] Disposing "DeviceAccessEntryPoint"
[2019-04-27 11:45:41.759 +00:00] [INF] Disposing "LibraryMonitorStartup"
[2019-04-27 11:45:41.759 +00:00] [INF] Disposing "AutomaticRestartEntryPoint"
[2019-04-27 11:45:41.760 +00:00] [INF] Disposing "ExternalPortForwarding"
[2019-04-27 11:45:41.761 +00:00] [INF] Disposing "LibraryChangedNotifier"
[2019-04-27 11:45:41.763 +00:00] [INF] Disposing "RecordingNotifier"
[2019-04-27 11:45:41.764 +00:00] [INF] Disposing "ServerEventNotifier"
[2019-04-27 11:45:41.767 +00:00] [INF] Disposing "StartupWizard"
[2019-04-27 11:45:41.767 +00:00] [INF] Disposing "UdpServerEntryPoint"
[2019-04-27 11:45:41.767 +00:00] [INF] Disposing "UserDataChangeNotifier"
[2019-04-27 11:45:41.768 +00:00] [INF] Disposing "DeviceManagerEntryPoint"
[2019-04-27 11:45:41.768 +00:00] [INF] Disposing "CollectionManagerEntryPoint"
[2019-04-27 11:45:41.768 +00:00] [INF] Disposing "ActivityLogEntryPoint"
[2019-04-27 11:45:41.771 +00:00] [INF] Disposing "DlnaEntryPoint"
[2019-04-27 11:45:41.772 +00:00] [INF] Disposing SsdpDevicePublisher
[2019-04-27 11:45:41.884 +00:00] [INF] Disposing PlayToManager
[2019-04-27 11:45:41.886 +00:00] [INF] Disposing DeviceDiscovery
[2019-04-27 11:45:41.890 +00:00] [INF] Disposing SsdpCommunicationsServer
[2019-04-27 11:45:41.892 +00:00] [INF] "SsdpCommunicationsServer" disposing _BroadcastListenSocket
[2019-04-27 11:45:41.896 +00:00] [INF] "SsdpCommunicationsServer" Disposing 2 sendSockets
[2019-04-27 11:45:41.896 +00:00] [INF] "SsdpCommunicationsServer" disposing sendSocket from "0.0.0.0"
[2019-04-27 11:45:41.896 +00:00] [INF] "SsdpCommunicationsServer" disposing sendSocket from "10.0.0.x"
[2019-04-27 11:45:41.897 +00:00] [INF] Disposing "Notifications"
[2019-04-27 11:45:41.900 +00:00] [INF] Disposing "EntryPoint"
[2019-04-27 11:45:41.904 +00:00] [INF] Starting new instance
[2019-04-27 11:45:41.906 +00:00] [INF] Executable: "/usr/lib/jellyfin/restart.sh"
[2019-04-27 11:45:41.906 +00:00] [INF] Arguments: "--restartpath=/usr/lib/jellyfin/restart.sh --ffmpeg=/usr/lib/jellyfin-ffmpeg/ffmpeg  "
 ```

**System (please complete the following information):**
 - OS: Ubuntu 18.04 (LXD container)
 - Kernel: 4.19.0-2-amd64
 - Browser: All
 - Jellyfin Version: 10.0.3.1
 - Reverse proxy: nginx

**Additional context**

Jellyfin is installed in an LXD container on a Debian 9 hypervisor. Also it has had the data directory restored from an older version of Jellyfin (10.2.1)

### PR DESCRIPTION
**Pull request by joshuaboniface**
**Changes**
The old restart script was buggy, as reported in #1320. This updated
script seems to work far more reliably and conforms to the existing
jellyfin-sudoers packages sudo configuration.

**Issues**
Fixes #1320

### CODE REVIEW

File: deployment/debian-package-x64/pkg-src/bin/restart.sh [hunk 1]
```diff
@@ -1,20 +1,36 @@
...
-  cmd=$(echo "$restart_cmd" | awk '{print $1}')
-  cmd_loc=$(command -v ${cmd})
-  if [[ -n "$cmd_loc" ]]; then
-    restart_cmd=$(echo "$restart_cmd" | sed -e "s%${cmd}%${cmd_loc}%")
-    echo "sleep 2; sudo $restart_cmd > /dev/null 2>&1" | at now > /dev/null 2>&1
-    exit 0
-  fi
-done
+cmd="$( get_service_command )"
+echo "Using system control command '$cmd' to restart Jellyfin..."
```
Comment (nvllsvm) [inline deployment/debian-package-x64/pkg-src/bin/restart.sh]: nit - sysv doesnt seem to be a command. I'd reword this to indicate what type of system was detected. Ex. `Detected systemd`, `Detected sysv`, `Detected upstart`.

Something like that.

File: deployment/fedora-package-x64/pkg-src/restart.sh [hunk 2]
```diff
@@ -1,6 +1,36 @@
...
+echo "Using system control command '$cmd' to restart Jellyfin..."
+case $cmd in
+    'systemctl')
+        echo "sleep 2; /usr/bin/sudo /usr/bin/systemctl restart jellyfin" | at now 
+        ;;
+    'service')
+        echo "sleep 2; /usr/bin/sudo /sbin/service jellyfin restart" | at now 
+        ;;
+    'sysv')
+        echo "sleep 2; /usr/bin/sudo /etc/init.d/jellyfin restart" | at now
```
Comment (nvllsvm) [inline deployment/fedora-package-x64/pkg-src/restart.sh]: Why `at now` instead of executing the commands directly?

File: deployment/debian-package-x64/pkg-src/bin/restart.sh (hunk 1 above)
Comment (joshuaboniface) [reply deployment/debian-package-x64/pkg-src/bin/restart.sh]: Fair point, just a small wording tweak. I'll change it.

File: deployment/fedora-package-x64/pkg-src/restart.sh (hunk 2 above)
Comment (joshuaboniface) [reply deployment/fedora-package-x64/pkg-src/restart.sh]: Because how JF restarts is bass-ackwards. It terminates the process like it was shutting down, then at the last second calls this script. However by that point the parent context is exiting, and it terminates the children, including this script. So we have to use this gross hack (as Emby did) to schedule the "start up" job externally from the script - `at` just seems like the easiest way to do so.

Comment (joshuaboniface) [reply deployment/fedora-package-x64/pkg-src/restart.sh]: I hate it too, believe me, but without getting into core code changes to perform a proper restart without invoking the external service manager, we're stuck with this one :-1:
